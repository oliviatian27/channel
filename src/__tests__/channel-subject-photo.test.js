import React from 'react';
import { shallow } from 'enzyme';

import ChannelSubjectPhoto from '../components/channel-list/channel-subject-photo';


describe('ChannelSubjectPhoto tests', () => {

  it('renders ChannelSubjectPhoto', () => {
    const subjectPhotoUrl = "some url";
    const wrapper = shallow(<ChannelSubjectPhoto subjectPhotoUrl={subjectPhotoUrl} />);

    // Expect the wrapper object to be defined
    expect(wrapper.find('.channel-subject-photo')).toBeDefined();
    expect(wrapper.find('.channel-subject-photo')).toMatchSnapshot();
  });

  it('displays the correct subjectPhoto based on props.subjectPhotoUrl', () => {
    const subjectPhotoUrl = "one";
    const wrapper = shallow(<ChannelSubjectPhoto subjectPhotoUrl={subjectPhotoUrl} />);

    // Expect the img src url to be props.subjectPhotoUrl
    expect(wrapper.find('img').prop("src")).toEqual(subjectPhotoUrl);

    // update the subjectPhotoUrl props
    wrapper.setProps({subjectPhotoUrl:"two"})

    expect(wrapper.find('img').prop("src")).toEqual("two");
  });

})
