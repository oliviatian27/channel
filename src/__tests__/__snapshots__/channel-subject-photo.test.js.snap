// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelSubjectPhoto tests renders ChannelSubjectPhoto 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelSubjectPhoto
      subjectPhotoUrl="some url"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="channel-subject"
          src="some url"
        />,
        "className": "channel-subject-photo",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "channel-subject",
          "src": "some url",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="channel-subject"
            src="some url"
          />,
          "className": "channel-subject-photo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "channel-subject",
            "src": "some url",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="channel-subject"
        src="some url"
      />,
      "className": "channel-subject-photo",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "channel-subject",
        "src": "some url",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="channel-subject"
          src="some url"
        />,
        "className": "channel-subject-photo",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "channel-subject",
          "src": "some url",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="channel-subject"
          src="some url"
        />,
        "className": "channel-subject-photo",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "channel-subject",
          "src": "some url",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
}
`;
